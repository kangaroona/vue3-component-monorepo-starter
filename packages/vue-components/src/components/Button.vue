<script setup lang="ts">
import { ref } from 'vue';

const props = withDefaults(
  defineProps<{
    text?: string;
  }>(),
  {
    text: 'World'
  }
);
const count = ref(0);
function clickHandler() {
  count.value += 2;
}
</script>

<template>
  <h1>{{ props.text }}</h1>
  <text class="count-text">{{ count }}</text>
  <button class="openx-button" @click="clickHandler">
    <slot />
  </button>
</template>
