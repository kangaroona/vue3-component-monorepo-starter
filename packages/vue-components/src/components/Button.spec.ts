import { mount, shallowMount } from '@vue/test-utils';
import { nextTick } from 'vue';
import Button from './Button.vue';

const text: string = 'text button';
describe('Button', () => {
  test('mount button', async () => {
    const wrapper = mount(Button, {
      props: {
        text
      }
    });
    expect(wrapper.text()).toContain(text);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('click button', async () => {
    const buttonClassName = '.openx-button';
    const countName = '.count-text';
    const wrapper = shallowMount(Button, {
      props: {
        text
      }
    });
    await wrapper.find(buttonClassName).trigger('click');
    await nextTick();
    expect(wrapper.find(countName).text()).toBe('1');
  });
});
