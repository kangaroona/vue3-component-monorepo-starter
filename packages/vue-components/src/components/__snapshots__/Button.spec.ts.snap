// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > mount button 1`] = `
"<h1>text button</h1>
<text class="count-text">0</text>
<button class="openx-button"></button>"
`;
